Source/Core/getStringFromTypedArray.js

```diff
@@ -42,20 +44,128 @@ define([
         return decoder.decode(view);
     };
 
+    var code_point = 0;
+    var bytes_seen = 0;
+    var bytes_needed = 0;
+    var lower_boundary = 0x80;
+    var upper_boundary = 0xBF;
+
     getStringFromTypedArray.decodeWithFromCharCode = function(view) {
+        code_point = bytes_needed = bytes_seen = 0;
+        lower_boundary = 0x80;
+        upper_boundary = 0xBF;
+
         var result = '';
         var length = view.length;
+        var i, cp;
+        var codePoints = [];
+        for (i = 0; i < length; ++i) {
```

Comment:
Would it be cleaner to move this loop into `utf8Handler` itself and have it return an array?  I think that would remove the need for all of the module scope variables and the "reset everything to 0" steps making the code easier to follow.  (But I could be wrong, up to you).